src/gitHub/GitHubRepositoryFileSelectStep.ts

```diff
@@ -0,0 +1,113 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the MIT License. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+import { AzureWizardPromptStep, IAzureQuickPickItem, nonNullProp } from "@microsoft/vscode-azext-utils";
+import type { IGitHubContext } from "./IGitHubContext";
+import { GetRepoContentReqParams, RepoContent, getRepositoryContent } from "./getRepositoryContent";
+
+type ContentPickData = {
+    traverse?: 'Up' | 'Down',
+    contentName: string;
+}
+
+/**
+ * A wizard step that allows the user to traverse through a remote GitHub repository,
+ * select a file, and store its relative path in context.
+ * @param fileName A RegExp or string that matches the file name to select
+ * @param contextKey The name of the key in the context object used to store the selected file path
+ * @param prompt The prompt to display to the user while selecting a file
+ */
+export class GitHubRepositoryFileSelectStep extends AzureWizardPromptStep<IGitHubContext> {
+    private cachedPicks: Map<string, IAzureQuickPickItem<ContentPickData>[]>;
+
+    private readonly fileName: RegExp;
+    private readonly contextKey: string;
+    private readonly promptPlaceHolder: string;
+
+    constructor (fileName: RegExp | string, contextKey: string, prompt: string) {
+        super();
+        this.fileName = fileName instanceof RegExp ? fileName : new RegExp(fileName);
+        this.contextKey = contextKey;
+        this.promptPlaceHolder = prompt;
+    }
+
+    public async prompt(context: IGitHubContext): Promise<void> {
+        // Reset cache when prompting in case the user has made breaking changes via the back button
+        this.cachedPicks = new Map();
+
+        let path: string = '';
+        while (true) {
+            const contentPickData = (await context.ui.showQuickPick(this.getPicks(context, path), { placeHolder: this.promptPlaceHolder })).data;
```

Comment:
Rather than` suppressPersistence` on every pick (down below) why not just set it as an option for the entire quickpick?

Reply:
I actually didn't realize there was a suppressPersistence option there 😂